### PR TITLE
[CI] Enable macOS in test matrix

### LINKED ISSUE #376: [Build] macOS CI
## What

Enable macOS CI

## ACs

- macOS CI runs in the test job
- Unsupported linters/sanitizers are skipped with a warning
- 10.15 minimum deployment target
- Issues logged for any subsequent steps to re-enable parts of the build tooling currently unsupported 

## Out of scope

 - Enabling missing tooling

### PR DESCRIPTION
**Pull request by foundrytom**
Closes #376

We disable sanitizers for now as we need more `asan` trickery and `ubsan` highlighted something either potentially interesting or a false positive. Added #381 to track this.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -24,8 +24,19 @@ jobs:
             vcvars: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"
           }
         - {
-            os: ubuntu-20.04
+            os: ubuntu-20.04,
+            sanitizers: ON
           }
+        - {
+            os: macos-10.15,
```
Comment (feltech) [inline .github/workflows/test.yml:31]: [fix] Remove TODO comment, above.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -24,8 +24,19 @@ jobs:
...
+            # asan needs the pre-load trick for python, ubsan seems to
+            # have false positives:
+            #   InfoDictionaryTest.cpp:346:25: runtime error:
+            #      call to function openassetio_v0_0_InfoDictionary_getBool
+            #      through pointer to incorrect function type
+            sanitizers: OFF
+          }
```
Comment (feltech) [inline .github/workflows/test.yml:38]: [question]] What do you think of the pseudo-JSON syntax (the `{ ... }`s and trailing `,`s)? I'm not a big fan in this case, it just seems to add unnecessary verbosity.

**Review by feltech — APPROVED** [review]
Just a minor fixup to remove an errant TODO, otherwise ready to merge.

File: .github/workflows/test.yml (hunk 2 above)
Comment (foundrytom) [reply .github/workflows/test.yml]: With you on that, updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/382/commits/2126aa6f873f184683486b46506798ec3fac4f6b).

**Pull request merged**